### PR TITLE
Fix exporting the remainder of the counters at the end of video

### PR DESCRIPTION
**Pull request by vahedi25411**

### CODE REVIEW

**Comment by ian-r-rose** [comment]
Thanks @vahedi25411! Looks like CI is complaining about the isort check.

**Comment by ian-r-rose** [comment]
@vahedi25411 This still doesn't seem to produce any counts in my testing. I just get an empty `counter.csv`:
```
Time,Pedestrian,Cyclist
```
I think this counter should also draw from the `search_objects` configurable when deciding which columns to output in the CSV.

**Comment by vahedi25411** [comment]
@ian-r-rose It should create an Excel file next to the video file, with the same name as the video's filename.

Yes, you are right. I should match the output columns with the selected object.

**Comment by ian-r-rose** [comment]
Ah, I missed that you changed the name of the output CSV. However, I still see the results being empty. My current test video (4 seconds long) counts 11 pedestrians and 1 cyclist, but I get an empty CSV.

**Comment by vahedi25411** [comment]
It is weired, I checked a video (19 seconds long) and it works fine, even when I stop the process at the 3th second, I can see the results in Excel file.

**Comment by ian-r-rose** [comment]
Weird, I am entering: `automated-walk-bike-counter --cli=True --file_name=a-trip-d
own-market.mp4 --save_periodic_counter=True`, am I missing something?

**Comment by vahedi25411** [comment]
I run it through GUI with the following parameters:

`save_periodic_counter=True` 
`periodic_counter_time=1`

Could you please check it on GUI?

**Comment by ian-r-rose** [comment]
Ah, interesting, it works when I run the GUI, but not when I run it via CLI.

**Comment by ian-r-rose** [comment]
It looks like `periodic_counter_time` is not getting a default, so if it is not set via CLI, then the counter fails.

**Comment by vahedi25411** [comment]
Yea, it is not related to CLI because both of the CLI and GUI use the same block of code to handle 15 minutes counters.

**Comment by ian-r-rose** [comment]
But when I run with GUI it works, and with CLI it doesn't.

**Comment by vahedi25411** [comment]
If you check it on GUI without `periodic_counter_time`, does it work?

**Comment by ian-r-rose** [comment]
No, then I get a `NoneType` error, because the `periodic_counter_time` isn't getting a default value.

**Comment by ian-r-rose** [comment]
A thought: might this be due to the export thread not finishing before the application exits?

**Comment by ian-r-rose** [comment]
Yep, if I `join()` the counter thread before exiting, then the CSV is produced as expected.

**Comment by vahedi25411** [comment]
Cool, do you commit this change?

**Comment by ian-r-rose** [comment]
I haven't yet, but I can push here if you like.

**Comment by ian-r-rose** [comment]
Okay, I've pushed the fix. Before we merge, can we 
1. Set a default value for `periodic_counter_interval`, and
2. Get the list of CSV exports from the configured values?

**Comment by vahedi25411** [comment]
Ok, no problem. I added the default value. I will add the code to make CSV based on the configured as soon as possible.

**Comment by ian-r-rose** [comment]
@vahedi25411 looks like you have not pushed the commit with the default value.

**Comment by vahedi25411** [comment]
@ian-r-rose I pushed the changes for making the excel file based on selected objects.

File: automated_walk_bike_counter/gui/controller/main_controller.py [hunk 1]
```diff
@@ -58,6 +58,11 @@ def update_video_canvas(self, filename, listener_object):
         tracker = ObjectTracker(self.mask)
         if self.video:
             tracker.video_filename = self.video.filename
+        tracker.valid_selected_objects = [
```
Comment (ian-r-rose) [inline automated_walk_bike_counter/gui/controller/main_controller.py]: Nice list comprehension. Why skip "bicycle"?

Comment (vahedi25411) [reply automated_walk_bike_counter/gui/controller/main_controller.py]: @ian-r-rose  Thanks. We have cyclists on this list so we don't need bicycles.
I fixed the issue and pushed the changes. Please use `--input_type=file` in your command.

**Comment by ian-r-rose** [comment]
I'm still seeing some odd behavior around the valid objects list. When I run without specifying the valid objects, it looks like it is counting `['person', 'cyclist', 'bicycle']`, which is weird, since the [defaults](https://github.com/CityOfLosAngeles/automated-walk-bike-counter/blob/2c4424a348ff68553ef9d06bc922d0c678815021/automated_walk_bike_counter/core/configuration.py#L128) are different from that. Is the configuration no longer being picked up correctly? I am also surprised to see `cyclist` and `bicycle` in the same list. I thought we were trying to skip one?

**Comment by vahedi25411** [comment]
The configuration is working correctly, and this is not a problem. `bicycle` is not in the list of objects that are selected by the user, but I need to have it in the list for my computations, that is why I add it to the list.

**Comment by ian-r-rose** [comment]
What is an example of a command-line/config file invocation that sets a different set of valid objects? It used to work, but now I am unable to get it to work.

What is the difference between `bicycle` and `cyclist`, and why does `bicycle` need to be added?

**Comment by vahedi25411** [comment]
This is the command line parameters that I use to run the program:

`--save_periodic_counter=True --periodic_counter_time=30 --file_name=<FILE_NAME> --cli=True --input_type=file`

The application detects cyclists by detecting bicycles.

**Comment by ian-r-rose** [comment]
Right, but I am talking about also configuring the list of valid objects (via either command line or config file). As far as I can tell, it is broken right now, but it was working before.

**Comment by vahedi25411** [comment]
I didn't change anything in the configuration file, but I am working on it to see what is the issue.

**Comment by vahedi25411** [comment]
I didn't change anything in the configuration file, but I am working on it to see what the issue is.

**Comment by ian-r-rose** [comment]
I guess I don't understand how the valid objects list is supposed to work.  It still doesn't make sense to me the cyclist and bicycle are in the same list, why not combine them into one item? I also thought that the algorithm used to have a working car option, but now I can't get it added to the list. What are the intended valid objects?

**Comment by vahedi25411** [comment]
For the cyclist, let me think more about it, and I will let you know.
For adding cars, do you have this problem on GUI?

**Comment by ian-r-rose** [comment]
I'm happy to think about those issues in a follow-up. For now, I think this works well, and should allow me to finish #55.

Thanks @vahedi25411!

**Pull request merged**